docs/data/charts/components/JSAnimationCustomization.tsx

```diff
@@ -0,0 +1,64 @@
+import { BarChart, BarLabelProps } from '@mui/x-charts/BarChart';
+import * as React from 'react';
+import { useAnimateBarLabel } from '@mui/x-charts/hooks';
+
+export default function JSAnimationCustomization() {
+  return (
+    <BarChart
+      xAxis={[{ scaleType: 'band', data: ['A', 'B', 'C'] }]}
+      series={[
+        {
+          type: 'bar',
+          data: [5, 17, 11],
+        },
+        {
+          type: 'bar',
+          data: [0, 12, 15],
+        },
+        {
+          type: 'bar',
+          data: [1, 3, 9],
+        },
+      ]}
+      width={300}
+      height={400}
+      barLabel="value"
+      slots={{ barLabel: AnimatedBarLabel }}
+    />
+  );
+}
+
+function AnimatedBarLabel(props: BarLabelProps) {
+  const {
+    seriesId,
+    dataIndex,
+    color,
+    isFaded,
+    isHighlighted,
+    classes,
+    xOrigin,
+    yOrigin,
+    x,
+    y,
+    width,
+    height,
+    layout,
+    skipAnimation,
+    ...otherProps
+  } = props;
+
+  const animatedProps = useAnimateBarLabel({
+    xOrigin,
+    x,
+    yOrigin,
+    y,
+    width,
+    height: -8,
```

Comment:
This feels a bit hacky because it requires knowledge of the implementation details of `useAnimateBarLabel`. 

I wonder if we should expose `useAnimate` after all (or maybe a higher-level version of it?)

Reply:
That's weird. Is it just to specify the initial value that you need this `height: -8`

If yes it might be easier to let user specify their own initial values.


Wy not having also the default initial height adapting to the layout like you do for x and y.
```js
const initialProps = {
  ...,
  height: props.layout === 'vertical' ? 0 : props.height
```

And if the -8 is here to hide the element, maybe the `opacity` could be an extra label to animate

Reply:
> Is it just to specify the initial value that you need this height: -8

It's not about the initial value, it's about the end value actually. I've since updated this code, but it's still a bit weird.

By default we render the bar labels in the middle of the bar. In this demo I'm moving them to the top of the bar and keeping the animation, so that's why I need this hack.

> If yes it might be easier to let user specify their own initial values.

That's another use case, but yeah, I'm considering exporting a slightly higher level version of `useAnimate`. 

I'll try to get a grasp of how something like that would look like.

Reply:
Created a higher-level `useAnimateRef` hook [here](https://github.com/mui/mui-x/commit/99e9abedadd37352a10d91c85ac6520c50e7a939). 

We might want to expose that one so that users can customize the animations using our "framework".

Reply:
> We might want to expose that one so that users can customize the animations using our "framework".

I agree. My point was more about the timeline. This PR could have a docs with a simpler example. But I struggle to find an interesting one. I thought about a bar label with a rectangle behind it but it does seems as hacky as the current one because the `useBarLabelAnimation` will only animate the x and y.

It seems that without hack, those hook are only to reproduce the exact same behavior as default components.

So maybe in this PR just documenting that user can use those hook to create their own version. And in another PR we will discuss the DX and

Reply:
> So maybe in this PR just documenting that user can use those hook to create their own version. And in another PR we will discuss the DX and

Sure. So you're saying that we should ship this as is and then in a separate PR export a higher-level hook to be customizable by users, right?

Reply:
[PoC here](https://github.com/mui/mui-x/pull/17162)